### PR TITLE
Implement the Dummy Task Details UI

### LINKED ISSUE #18: Create view single task component

### PR DESCRIPTION
**Pull request by kavisha-perera**
## Purpose

The purpose of this PR is to fix https://github.com/we-huddle/server/issues/18

## Goals

Implement a UI to view task details (with dummy data).

## Approach

Created a component called TaskDetails

### CODE REVIEW

File: src/scenes/TaskDetails/index.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import {Profile} from "../../types/Profile";
+import {useContext} from "react";
+import userContext from "../../types/UserContext";
+import {Badge, Button, Card} from "flowbite-react";
+import {HiCheck, HiPencil, HiTrash} from "react-icons/hi";
+
+
+interface TaskViewProps {
```
Comment (Gravewalker666) [inline src/scenes/TaskDetails/index.tsx]:
```suggestion
interface TaskDetailsViewProps {
```

File: src/scenes/TaskDetails/index.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    isAgentView: boolean,
+  }
+
+function TaskDetails({ isAgentView }: TaskViewProps){
```
Comment (Gravewalker666) [inline src/scenes/TaskDetails/index.tsx]:
```suggestion
function TaskDetailsView({ isAgentView }: TaskViewProps){
```
Lets rename this as `TaskDetailsView` to keep the naming convention consistent, make sure you rename the directory name as well, (hint: use refactor on IntelliJ)

File: src/scenes/TasksView/components/TaskCard/index.tsx [hunk 3]
```diff
@@ -32,9 +35,12 @@ function TaskCard({task, completed}: TaskCardProps) {
...
           )}
         </div>
+       
       </Card>
     </div>
   );
 }
 
 export default TaskCard;
+
```
Comment (Gravewalker666) [inline src/scenes/TasksView/components/TaskCard/index.tsx:41]: Some unnecessary changes, probably by mistake you've edited the file, lets undo these changes

File: src/scenes/TasksView/index.tsx [hunk 4]
```diff
@@ -231,17 +233,39 @@ function TasksView({ isAgentView }: TaskViewProps) {
...
+                key={task.id}
+                task={task}
+                completed={!isAgentView && completedTaskIdList.includes(task.id)}
+              />
+              </Link>
+           );
+         })}
+       </div>
+        )}
+
```
Comment (Gravewalker666) [inline src/scenes/TasksView/index.tsx]: I see what you've tried to do here, to send them to the correct details page depending on their role, 
This is a pretty good working solution and there's nothing wrong with it, But we can do the same thing with less code, we can just keep one of these tasks.map functions and use the link,
```javascript
<Link to={`${window.location.pathname/details}`}>
```
to navigate them to the correct page

**Review by Gravewalker666 — CHANGES_REQUESTED** [review]

File: src/scenes/TasksView/components/TaskCard/index.tsx (hunk 3 above)
Comment (Gravewalker666) [reply src/scenes/TasksView/components/TaskCard/index.tsx:41]: This entire file*

Comment (kavisha-perera) [reply src/scenes/TasksView/components/TaskCard/index.tsx:41]: This is now fixed

File: src/scenes/TaskDetails/index.tsx (hunk 2 above)
Comment (kavisha-perera) [reply src/scenes/TaskDetails/index.tsx]: All the instances of TaskViewProps and TaskDetails were renamed to TaskDetailsViewProps and TaskDetailsView respectively.

**Review by Gravewalker666 — APPROVED** [review]
Awesome work @kavisha-perera really appreciate the effort you put in, good job!
